### PR TITLE
lib/systemd: expose systemd unit translation helpers

### PR DESCRIPTION
**Pull request by Malix-Labs**
### Description

This PR extracts Home Manager's internal systemd unit translation logic (`toHmIni`, `toHmIniSocket`, `normalizeTarget`, `envToList`, `pickSection`, `installSection`) into a public library module `lib.hm.systemd` (`modules/lib/systemd.nix`).

It enables flake authors maintaining both `nixosModules` and `homeModules` to easily convert and reuse NixOS-style unit definitions in Home Manager without duplicating unit metadata and scripts.

Additionally, 5 service modules (`comodoro`, `exo`, `local-ai`, `ollama`, `radicle`) have been refactored to use `lib.hm.systemd.envToList` instead of ad-hoc `mapAttrsToList` string formatting.

An alternative is to make Home Manager options to have the same schema as NixOS' ones (or the other way around)
I find it preferable, but don't know if this is possible nor desirable

### Checklist

- [x] Change is backwards compatible.
- [x] Code formatted with `nix fmt` or `nix-shell -A dev --run treefmt`.
- [ ] Code tested through `nix build .#test-all` or a targeted `nix run .#tests -- <pattern>`.
- [ ] Test cases updated/added.
- [x] Commit messages are formatted like `{component}: {description}`.

### CODE REVIEW

**Comment by Malix-Labs** [comment]
That PR is more architectural than it is specific to those packages

I think the relevant people are @rycee, @khaneliman, @KiaraGrouwstra

**Review by KiaraGrouwstra — APPROVED** [review]
this looks much cleaner already compared to my original there - thanks!

**Comment by Malix-Labs** [comment]
Btw most of the additional lines are due to nixdoc comments. the rest is because splitting files duplicate some boilerplate of course

**Comment by Malix-Labs** [comment]
That PR is good only if we want to keep using those options;

However, most of it wouldn't be needed if home manager mirrored NixOS' options, right?

Is there a reason why NixOS options for systemd don't match the ini values? 
I know it doesn't just merely generate the end result file and adds some typing / checks on top of it, but can't it just do this AND mirror the systemd ini schema anyway?

**Comment by KiaraGrouwstra** [comment]
i wouldn't know the answer there.

that said, i'm trying to add user services there as well (https://github.com/NixOS/nixpkgs/pull/524517), but also wanna make things more principled upstream (https://github.com/NixOS/nixpkgs/pull/540863).

**Comment by Malix-Labs** [comment]
That's cool, maybe this PR shouldn't be merged, then?

And also, is there someone I can @ mention there who would be able to answer https://github.com/nix-community/home-manager/pull/9693#issuecomment-5036723195 ?

**Comment by Malix-Labs** [comment]
@KiaraGrouwstra when your nixpkgs modular services framework in NixOS/nixpkgs#524517 is merged and Home Manager adopts it as a first class citizen by allowing running a modular service directly in HM, lib.hm.systemd.toHmIni and such will be obsolete (for the better)

So if this is planned, and doable in the "short" term, I don't think this PR is good to be merged since it will create soon-to-be-deprecated libs that users could depend on (it's not the end of the world since we can create the usual deprecation warning anyway, but still)

**Comment by KiaraGrouwstra** [comment]
> is there someone I can @ mention there who would be able to answer https://github.com/nix-community/home-manager/pull/9693#issuecomment-5036723195 ?

perhaps @roberth would know more on systemd use at nixpkgs? otherwise it's probably gonna be a git blame.

> That's cool, maybe this PR shouldn't be merged, then?

> @KiaraGrouwstra when your nixpkgs modular services framework in https://github.com/NixOS/nixpkgs/pull/524517 is merged and Home Manager adopts it as a first class citizen by allowing running a modular service directly in HM, lib.hm.systemd.toHmIni and such will be obsolete (for the better)

> So if this is planned, and doable in the "short" term, I don't think this PR is good to be merged since it will create soon-to-be-deprecated libs that users could depend on (it's not the end of the world since we can create the usual deprecation warning anyway, but still)

i'll try and get a read on where the room is at on the next modular service meeting, maybe that could help.

File: modules/lib/systemd.nix [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  };
+in
+{
+  inherit
+    unitAttrKeys
+    normalizeTarget
+    normalizeTargets
+    pickSection
+    envToList
+    installSection
```
Comment (khaneliman) [inline modules/lib/systemd.nix:133]: issue (blocking): fix directive casing before publishing this API

`pickSection` uses `lib.toSentenceCase`, which emits `Bindsto`, `Partof`, `Onfailure`, and `Onsuccess`; systemd will not apply the intended dependency directives. Use an explicit key mapping and add direct tests for the exported translators, or keep them private.

File: modules/services/radicle.nix [hunk 2]
```diff
@@ -34,7 +32,7 @@ let
   radicleHome = config.home.homeDirectory + "/.radicle";
 
   gitPath = [ "PATH=${getBin pkgs.gitMinimal}/bin" ];
-  env = attrs: (mapAttrsToList (generators.mkKeyValueDefault { } "=") attrs) ++ gitPath;
+  env = attrs: (lib.hm.systemd.envToList attrs) ++ gitPath;
```
Comment (khaneliman) [inline modules/services/radicle.nix:35]: todo (non-blocking): pin the corrected environment semantics

This switch intentionally improves behavior: null entries are omitted, while declared-valid path and package values no longer abort evaluation. Add a Radicle service test covering null omission and path/package serialization so this user-visible change stays stable.

**Comment by teto** [comment]
> So if this is planned, and doable in the "short" term, I don't think this PR is good to be merged since it will create soon-to-be-deprecated libs that users could depend on (it's not the end of the world since we can create the usual deprecation warning anyway, but still)

I wouldn't hold the PR for this as modular services stuff seem controversial and might take some time to get merged. 
If it gets merged before 26.11 we dont care about backwards compat, we can scrap api etc otherwise we can alias or update the expressions to become wrappers of nixpkgs' new functions.

**Review by teto — APPROVED** [review]

**Comment by Malix-Labs** [comment]
@teto ok then

**Review by khaneliman — COMMENTED** [review]
The extraction is mechanically sound, but the public translator currently mis-cases several multiword systemd directives and has no direct API coverage. The Radicle environment conversion looks corrective and is non-blocking once its intended behavior is tested.

File: modules/services/radicle.nix (hunk 2 above)
Comment (Malix-Labs) [reply modules/services/radicle.nix:35]: > Add a Radicle service test covering null omission and path/package serialization so this user-visible change stays stable.

Should this be handled in this PR or should I let this to Radicle's module maintainers?

Comment (khaneliman) [reply modules/services/radicle.nix:35]: It'd be appreciated, but not required.

File: modules/lib/systemd.nix (hunk 1 above)
Comment (Malix-Labs) [reply modules/lib/systemd.nix:133]: done

File: modules/services/radicle.nix (hunk 2 above)
Comment (Malix-Labs) [reply modules/services/radicle.nix:35]: done
